core/cmd/lambda/main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/kislerdm/diagramastext/core"
+)
+
+type Request struct {
+	Prompt string `json:"prompt"`
+}
+
+func main() {
+	clientOpenAI, err := core.NewOpenAIClient(
+		core.ConfigOpenAI{
+			Token:       os.Getenv("OPENAI_API_KEY"),
+			MaxTokens:   mustParseInt(os.Getenv("OPENAI_MAX_TOKENS")),
+			Temperature: mustParseFloat32(os.Getenv("OPENAI_TEMPERATURE")),
+		},
+	)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	clientPlantUML := core.NewPlantUMLClient()
+
+	lambda.Start(
+		func(ctx context.Context, req events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
+			var r Request
+			if err := json.Unmarshal([]byte(req.Body), &r); err != nil {
+				return events.APIGatewayProxyResponse{
+					StatusCode: http.StatusBadRequest,
+				}, err
+			}
+
+			graph, err := clientOpenAI.Do(ctx, r.Prompt)
+			if err != nil {
+				return events.APIGatewayProxyResponse{
+					StatusCode: http.StatusInternalServerError,
+					Body:       "could not recognise diagram description",
+				}, err
+			}
+
+			svg, err := clientPlantUML.Do(context.Background(), graph)
```

Comment:
**question** why not the same `ctx` instead of this `context.Background()`?

Reply:
ah, it's a copy-paste typo - good catch! thank you!